File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        res, summ = nums[0], nums[0]
        for i in range(1, len(nums)):
            if nums[i - 1] < nums[i]:
                summ += nums[i]
            else:
                res = max(res, summ)
                summ = nums[i]
        return max(res, summ)

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.java
class Solution {
    public int maxAscendingSum(int[] nums) {
        int res = nums[0],sum = nums[0];
        for(int i = 1;i < nums.length;i++){
            if(nums[i - 1] < nums[i]){
                sum += nums[i];
            }else{
                res = Math.max(res,sum);
                sum = nums[i];
            }
        }
        return Math.max(res,sum);
    }
}